src/Sylius/Component/Core/OrderProcessing/OrderShipmentProcessor.php

```diff
@@ -69,6 +69,10 @@ public function process(BaseOrderInterface $order)
             return;
         }
 
+        foreach ($shipment->getUnits() as $unit) {
```

Comment:
Side note: This flow of removing all the entities from the collection and adding some of them back does not work with `orphanRemoval` enabled. Fortunately, it's explicitly disabled in the `Shipment` mapping, but still, it's something we should be aware of.

It's all right by now, but we can make it more reliable by performing only the necessary operations like in https://github.com/Sylake/SyliusConsumerPlugin/blob/master/src/Projector/ProductProjector.php#L216-L239.